demos/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c

```diff
@@ -0,0 +1,497 @@
+    double xIncomingTemperature;
+    uint32_t ulVersion;
+    bool xWasMaxTemperatureChanged = false;
+
+    xResult = prvProcessProperties( pxMessage, eAzureIoTHubClientPropertyWritable, &xIncomingTemperature, &ulVersion );
+
+    if( xResult == eAzureIoTSuccess )
+    {
+        prvUpdateLocalProperties( xIncomingTemperature, ulVersion, &xWasMaxTemperatureChanged );
+        *pulWritablePropertyResponseBufferLength = prvGenerateAckForIncomingTemperature(
+            xIncomingTemperature,
+            ulVersion,
+            pucWritablePropertyResponseBuffer,
+            ulWritablePropertyResponseBufferSize );
+    }
+    else
+    {
+        LogError( ( "There was an error processing incoming properties: result 0x%08x", xResult ) );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+uint32_t ulHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                          uint32_t * pulResponseStatus,
+                          uint8_t * pucCommandResponsePayloadBuffer,
+                          uint32_t ulCommandResponsePayloadBufferSize )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lCommandNameLength;
+    int32_t ulCommandResponsePayloadLength;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               ( const char * ) pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = sizeof( sampleazureiotCOMMAND_MAX_MIN_REPORT ) - 1;
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, ( const char * ) pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+
+        /*Initialize the reader from which we pull the "since". */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        /* Initialize the JSON writer with a buffer to which we will write the response payload. */
+        xResult = AzureIoTJSONWriter_Init( &xWriter, pucCommandResponsePayloadBuffer, ulCommandResponsePayloadBufferSize );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        /* Read from the writer the "since" value and use it to construct the response payload in the writer. */
+        xResult = prvInvokeMaxMinCommand( &xReader, &xWriter );
+
+        if( xResult == eAzureIoTSuccess )
+        {
+            ulCommandResponsePayloadLength = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+            *pulResponseStatus = AZ_IOT_STATUS_OK;
+        }
+        else
+        {
+            LogError( ( "Error generating command payload: result 0x%08x", xResult ) );
+
+            *pulResponseStatus = 501;
+            ulCommandResponsePayloadLength = sizeof( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) - 1;
```

Comment:
>ulCommandResponsePayloadLength [](http://example.com/codeflow?start=12&length=30)

please check the buffer size passed to this function.

Reply:
Added for both default implementation here and for esp32 azure iot kit implementation.